src/entity.rs

```diff
@@ -124,8 +124,16 @@ impl PartialOrd<Self> for CharsetMatch {
         let coherence_difference = (coherence_a - coherence_b).abs();
 
         // Below 1% difference --> Use Coherence
-        if mess_difference < 0.01 && coherence_difference > 0.02 {
-            return coherence_b.partial_cmp(&coherence_a);
+        if mess_difference < 0.01 {
+            if coherence_difference > 0.02 {
+                return coherence_b.partial_cmp(&coherence_a);
+            }
+            let multibyte_usage_a = self.multi_byte_usage();
+            let multibyte_usage_b = other.multi_byte_usage();
+            let multibyte_usage_delta = (multibyte_usage_a - multibyte_usage_b).abs();
+            if multibyte_usage_a.abs() > f32::EPSILON && multibyte_usage_b.abs() > f32::EPSILON {
```

Comment:
maybe it should be `multibyte_usage_delta > f32:EPSILON ` ?

Reply:
https://github.com/nickspring/charset-normalizer-rs/pull/25/commits/c3136ec4f44573adc693ece90d1deb213febc909#r1342504166

unfortunately that does not work either.

Reply:
What do you mean "does not work"?
Accuracy: 97.1% that's OK.

Reply:
Ah I meant that it does not reach 97.1%

Reply:
Both versions have 97.1%, but I wrote just about logic - we should compare multibyte_usage values if they are different only.

Reply:
Oh. Now I understand.
There must be an issue on my end! That is unfortunate.
Reverted to  `multibyte_usage_delta > f32:EPSILON`
in d4eae53